transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -201,7 +209,8 @@ private int epollWait(boolean oldWakenUp) throws IOException {
         long currentTimeNanos = System.nanoTime();
         long selectDeadLineNanos = currentTimeNanos + delayNanos(currentTimeNanos);
         for (;;) {
-            long timeoutMillis = (selectDeadLineNanos - currentTimeNanos + 500000L) / 1000000L;
+            long timeoutMillis = Math.min(MAX_EPOLL_WAIT_TIMEOUT,
```

Comment:
Can someone explain this sleep logic here? The native code also ensures it sleeps the full time if EINTR pops up.

I'm wondering if I need to do something similar in kqueue https://github.com/netty/netty/pull/6000/files#diff-54822e50acc8dd1c0bd8053332e65659R141